tree/criterion_interface.hpp

```diff
@@ -0,0 +1,70 @@
+#ifndef MACHINELEARNING_TREE_CRITERION_HPP
```

Comment:
There are many reasons to now follow the .h to all header files. I guess this is a topic to discuss in the next meeting. Follow the link explaining some differences of header files: [Link](http://stackoverflow.com/questions/152555/h-or-hpp-for-your-class-definitions)